### ISSUE TITLE
Add current_law_policy.json parameter for chained CPI indexing

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#1621]
As discussed in https://github.com/open-source-economics/Tax-Calculator/issues/1612, Tax-Calculator users have the capability to switch to chained CPI by modifying the _ACPIU grow factor. I'd like to propose adding a parameter to current_law_policy.json for doing the same. 

`Current_law_policy.json` is a more natural place to include what is essentially a policy parameter, and it would allow us to make the parameter available to TaxBrain GUI users. 

cc @codykallen @martinholmer

**Comment by martinholmer** [comment]
In the social security policy world they’ve been discussing chained CPI reforms for years (as a way to reduce benefits).  In that work the policy parameter is a cpi_offset, which is zero under current law.  To simulate a chained CPI reform user sets cpi_offset to, say, -0.005 if they think the reform will reduces the rate of price inflation by one-half of a percentage point. 

Does something like this approach make sense?

What are people (like TPC and CBO) assuming for the value of cpi_offset?

**Comment by codykallen** [comment]
@MattHJensen asked

> What are people (like TPC and CBO) assuming for the value of cpi_offset?

I don't know with certainty, but I suspect they are substituting the CBO forecasts for chained CPI for the CBO forecasts of CPI-U, both of which can be found in their 10-Year Economic Projections.

**Comment by MattHJensen** [comment]
A cpi_offset parameter makes sense to me. 

This [CBO report](https://www.cbo.gov/publication/44089) says:

> The chained CPI grows more slowly than the traditional CPI does: an average of about **0.25 percentage points** more slowly per year over the past decade.

**Comment by codykallen** [comment]
Given the potential for tax proposals to apply chained CPI inflation to some parameters in some years but not others, it would be useful to be able to change the inflation adjustment mechanism more directly than with `growdiff`. Ideally, we would be able to change the `"cpi_inflated"` term for each parameter in current_law_policy.json as part of a reform dictionary, instead of only being able to change the `"value"` item.

**Comment by martinholmer** [comment]
@codykallen said in issue #1621:
>  Ideally, we would be able to change the "cpi_inflated" term for each parameter in `current_law_policy.json` as part of a reform dictionary, instead of only being able to change the "value" item.

If I'm understanding you correctly, any user can do that now (and has been able to do that from the beginning of the project).  The "cpi_inflated" term is boolean and can be switched from true to false or from false to true in any calendar year.  Here's an example taken from the `test_policy.py` file:
```
 # specify multi-year reform using a dictionary of year_provisions dicts
    reform = {
        2015: {
            '_CTC_c': [2000]
        },
        2016: {
            '_EITC_c': [[900, 5000, 8000, 9000]],
            '_II_em': [7000],
            '_SS_Earnings_c': [300000]
        },
        2017: {
            '_SS_Earnings_c': [500000], '_SS_Earnings_c_cpi': False
        },
        2019: {
            '_EITC_c': [[1200, 7000, 10000, 12000]],
            '_II_em': [9000],
            '_SS_Earnings_c': [700000], '_SS_Earnings_c_cpi': True
        }
    }
```
And, of course, this sort of indexing reform can be done in JSON reform files as well.

**Comment by MattHJensen** [comment]
> @martinholmer, I was not aware we could change the CPI inflation boolean in that manner, but that makes things easier. Then it would seem that the next step to implementing the chained CPI inflation correctly would add another growth factor, such as ACCPI, that would act as a chained CPI growth factor. That could be accompanied by switching the "cpi_inflated" boolean term to something like inflation_adjustment, with values for not inflation adjusted, CPI adjusted, or chained CPI adjusted.

This does seem like an elegant solution, but it would break compatibility with the TaxBrain front end. It would be nice if this we did not require this as a short term solution. 

![image](https://user-images.githubusercontent.com/8301092/32506200-f7395e4e-c3b1-11e7-80cf-7804e29787f7.png)

@GoFroggyRun, what would be the effort level to make CPI flag show a drop down with multiple options when you click on it? 

cc @hdoupe.

**Comment by MattHJensen** [comment]
Given @codykallen's agreement with this:

> the majority of important provisions adopt the new chained cpi in 2018

I would still advocate for the first step to be adding a `cpi_offset` to `current_law_policy.json` term as @martinholmer proposed https://github.com/open-source-economics/Tax-Calculator/issues/1621#issuecomment-341749772.

**Comment by GoFroggyRun** [comment]
@MattHJensen asked: 
>@GoFroggyRun, what would be the effort level to make CPI flag show a drop down with multiple options when you click on it?

Are we trying to substitute the CPI flag buttons for all parameters with drop downs? The process shouldn't be technically difficult as long as I have walked through the process once. There seems to be more effort on the backend.

**Comment by martinholmer** [comment]
Issue #1621 has been resolved by the merge of pull request #1667.
